express-post-json/index.js

```diff
@@ -0,0 +1,36 @@
+const express = require('express');
+const app = express();
+
+let nextId = 1;
+const grades = {};
+
+app.listen(3000, () => {
+  // eslint-disable-next-line
+  console.log('Express server listening on port 3000');
+});
+
+const gradesArr = [];
+app.get('/api/grades', function (req, res) {
+  for (const key in grades) {
+    gradesArr.push(grades[key]);
+  }
+  res.json(gradesArr);
+});
+
+app.use(express.json());
+
+app.post('/api/grades', function (req, res) {
+  res.status(201).json({
+    course: req.body.course,
+    name: req.body.name,
+    score: req.body.score,
+    nextId: nextId
+  });
+  grades.nextId = {
+    course: req.body.course,
+    name: req.body.name,
+    score: req.body.score,
+    nextId: nextId
+  };
+  nextId++;
```

Comment:
each grade stored in the data model should have an `id` property (not `nextId`).

also, it's best to store the data _first_ before sending a success response.